.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,14 @@
+on:
+  push:
+    paths:
+      - '**/*.sh'
+      - 'bin/*'
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ludeeus/action-shellcheck@1.1.0
```

Comment:
Most are POSIX and should be tested with `dart` as the shell language. Some are bash and should be tested with bash.

Reply:
I think that can be done on a per-file basis with a `shellcheck` directive. https://www.shellcheck.net/wiki/Directive#shell. Can definitely add it.